### PR TITLE
Remove padding while calculating unitBarWidth; invalidateScaleBar while applySettings

### PR DESCRIPTION
**Pull request by Chaoba**
Instead of minus `INTERNAL_PADDING_DP` while calculating unitBarWidth, now add `INTERNAL_PADDING_DP` to the width of scalebar to avoid the issue that scalebar is a litter shorter than it should be.

Invoke `invalidateScaleBar()` while updating scalebar settings to make sure it updates to the new state.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt [hunk 1]
```diff
@@ -364,6 +364,5 @@ class ScaleBarImpl : ScaleBar, View {
     internal const val MSG_RENDER_CONTINUOUS = 1
     internal const val DEFAULT_MAPVIEW_WIDTH = 0F
     internal const val DEFAULT_PIXEL_RATIO = 1.0F
-    internal const val INTERNAL_PADDING_DP = 10
```
Comment (pengdev) [inline plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt:367]: Do you remember why this was introduced?

File: plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt [hunk 2]
```diff
@@ -67,7 +67,7 @@ class ScaleBarImpl : ScaleBar, View {
 
   @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
   internal fun calculateWidthAndHeight(): Pair<Float, Float> {
-    val width = mapViewWidth * settings.ratio
+    val width = mapViewWidth * settings.ratio + INTERNAL_PADDING_DP * pixelRatio
```
Comment (kiryldz) [inline plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt:70]: btw isn't `settings.ratio` same as pixelRatio?

File: plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt (hunk 1 above)
Comment (Chaoba) [reply plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt]: We can add tests while pixel match test is ready.

**Review by ZiZasaurus — APPROVED** [review]
Screen Shot 2022-05-06 at 11 42 38 AM
LGTM!

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ Mapbox welcomes participation and contributions from everyone.
 
 ## Bug fixes 🐞
 * Enable two finger pan gesture. ([#1280](https://github.com/mapbox/mapbox-maps-android/pull/1280))
+* Fix a bug that scale bar is shorter than it should be; trigger `invalidateScaleBar` while updating settings to make scale keep the latest status. ([#1336](https://github.com/mapbox/mapbox-maps-android/pull/1336))
```
Comment (yunikkk) [inline CHANGELOG.md:11]: Imo too verbose, `Fix a bug that scale bar is shorter than it should be` is good enough, rest details are in the PR

Comment (Chaoba) [reply CHANGELOG.md:11]: I'm using `;` to divide reset is to fix https://github.com/mapbox/mapbox-maps-android-internal/issues/1091

Comment (yunikkk) [reply CHANGELOG.md:11]: Ok, got it!

**Review by yunikkk — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Pull request merged**